app/graphql/types/topic_type.rb

```diff
@@ -24,6 +25,16 @@ def creator
       # object.first_post.creator
     end
 
+    def solved
+      BatchLoader::GraphQL.for(object.id).batch(default_value: false) do |topic_ids, loader|
+        Post.where(topic_id: topic_ids, solution: true).each do |post|
+          loader.call(post.topic_id, true)
+        end
```

Comment:
@mahesh-krishnakumar I think it should be possible to avoid instantiating full `Post` objects here by _plucking_ only the information that we care about:

```ruby
Post.where(topic_id: topic_ids, solution: true).distinct(:topic_id).pluck(:topic_id).each do |topic_id|
  loader.call(topic_id, true)
end
```